pycbc/workflow/grb_utils.py

```diff
@@ -365,3 +400,177 @@ def get_sky_grid_scale(
+                       segment[0], segment[1]-segment[0])
+            out_file = File(ifo_tag, 'trig_combiner', segment,
+                            file_url=os.path.join(out_dir, out_name))
+            node.add_output(out_file)
+            output_files.append(out_file)
+
+        return node, output_files
+
+
+class PycbcGrbTrigClusterExecutable(Executable):
+    """ The class responsible for creating jobs
+    for ''pycbc_grb_trig_cluster''.
+    """
+
+    current_retention_level = Executable.ALL_TRIGGERS
+
+    def __init__(self, cp, name):
+        super().__init__(cp=cp, name=name)
+
+    def create_node(self, in_file, out_dir):
+        node = Node(self)
+        node.add_input_opt("--trig-file", in_file)
+        # Determine output file name
+        ifotag, filetag, segment = filename_metadata(in_file.name)
+        start, end = segment
+        out_name = "{}-{}_CLUSTERED-{}-{}.h5".format(ifotag, filetag,
+                                                     start, end-start)
+        out_file = File(ifotag, 'trig_cluster', segment,
+                        file_url=os.path.join(out_dir, out_name))
+        node.add_output(out_file)
+
+        return node, out_file
+
+
+class PycbcGrbInjFinderExecutable(Executable):
+    """The class responsible for creating jobs for ``pycbc_grb_inj_finder``
+    """
+    current_retention_level = Executable.ALL_TRIGGERS
+
+    def __init__(self, cp, exe_name):
+        super().__init__(cp=cp, name=exe_name)
+
+    def create_node(self, inj_files, inj_insp_files,
+                    out_dir, tags=None):
+        if tags is None:
+            tags = []
+        node = Node(self)
+        node.add_input_list_opt('--input-files', inj_insp_files)
+        node.add_input_list_opt('--inj-files', inj_files)
+        ifo_tag, desc, segment = filename_metadata(inj_files[0].name)
+        desc = '_'.join(desc.split('_')[:-1])
+        out_name = "{}-{}_FOUNDMISSED-{}-{}.h5".format(
+            ifo_tag, desc, segment[0], abs(segment))
+        out_file = File(ifo_tag, 'inj_finder', segment,
+                        os.path.join(out_dir, out_name), tags=tags)
+        node.add_output(out_file)
+        return node, out_file
+
+
+class PycbcGrbInjCombinerExecutable(Executable):
+    """The class responsible for creating jobs ``pycbc_grb_inj_combiner``
+    """
+    current_retention_level = Executable.ALL_TRIGGERS
+
+    def __init__(self, cp, exe_name):
+        super().__init__(cp=cp, name=exe_name)
+
+    def create_node(self, input_file, out_dir, ifo_tag, segment, tags=None):
+        if tags is None:
+            tags = []
+        node = Node(self)
+        node.add_input_opt('--input-files', input_file)
+        out_name = input_file.name[:-3] + '-FILTERED.h5'
+        out_file = File(ifo_tag, 'inj_combiner', segment,
+                        os.path.join(out_dir, out_name), tags=tags)
+        node.add_output_opt('--output-file', out_file)
+        node.add_opt('--max-inclination', 30)
```

Comment:
This sounds like something that we absolutely want in a config file and not hardcoded here, maybe add a FIXME comment.